app/Http/Controllers/PostController.php

```diff
@@ -1,18 +1,49 @@
 <?php
 
 namespace App\Http\Controllers;
+use App\Location;
+use App\Post;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Storage;
 
 class PostController extends Controller
 {
 
-    public function create()
+    public function create($id)
     {
-        return view('posts.create');
+        //$location = Location::with('posts')->findOrFail($id);
+
+        return view('posts.create')->with("location", $id);
```

Comment:
Include the full location object or call this variable `location_id`. This is a recipe for disaster and confusion where something explodes because you expected a location and got an integer.